File: backend/src/main/java/ca/metricalsky/winston/entity/view/ChannelVideoStatisticsView.java
package ca.metricalsky.winston.entity.view;

import java.time.OffsetDateTime;


public interface ChannelVideoStatisticsView {

    String getChannelHandle();

    String getChannelId();

    OffsetDateTime getChannelPublishedAt();

    Integer getChannelVideoCount();

    Integer getVideoCount();

    OffsetDateTime getLatestVideoPublishedAt();
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/view/ChannelVideoStatisticsView.kt
package ca.metricalsky.winston.database.view

import java.time.OffsetDateTime

interface ChannelVideoStatisticsView {

    val channelHandle: String
    val channelId: String
    val channelPublishedAt: OffsetDateTime
    val channelVideoCount: Int?
    val videoCount: Int
    val latestVideoPublishedAt: OffsetDateTime
}
